scipy/stats/_continuous_distns.py

```diff
@@ -3667,12 +3695,15 @@ def fit(self, data, *args, **kwds):
         # The fit method of `gumbel_r` can be used for this distribution with
         # small modifications. The process to do this is
         # 1. pass the sign negated data into `gumbel_r.fit`
+        #    - if the location is fixed, it should also be negated.
         # 2. negate the sign of the resulting location, leaving the scale
         #    unmodified.
         # `gumbel_r.fit` holds necessary input checks.
 
+        if kwds.get('floc') is not None:
+            kwds['floc'] = -kwds['floc']
```

Comment:
Tests weren't picking up that `gumbel_r` wasn't working correctly. Can you add a test that would have failed in `main`?

Actually... sorry to change my mind, but maybe better yet would be to make a separate PR for this change. I can merge that one myself really quickly.

Reply:
This was not detected in main because the current `check_fit_args_fix` fixed `floc` to `0`, so negative or positive errors aren't tested.

Unfortunately since the data was generated with loc=0 and scale=1, it doesn't work to fix the location to 1 for every distribution, so that's out. Adding another test with nonzero floc and possibly non-one scale (with appropriately generated data) seems like the way to go.

Reply:
Ok adding a specialized test seems fine.

Reply:
See gh-16101